File: dataStructure/queues/QueueWithLinkedList.java

package queues;

public class QueueWithLinkedList {
    private Node first;
    private Node last;
    private int length;

    public QueueWithLinkedList() {
        this.length = -1;
    }

    public Integer peek(){
        return first.value;
    }

    public void enqueue(Integer value){
        Node newNode = new Node(value);
        if(isEmpty()) {
            first = newNode;
            last = newNode;
        } else {
            newNode.next = last;
            last = newNode; 
        }

        length++;
    }

    public void dequeue(){
        if(isEmpty()) return;

        Node currentNode = last;

        while(currentNode.next != first) {
            currentNode = currentNode.next;
        }

        first = currentNode;
        first.next = null;
        length--;
    }

    public boolean isEmpty(){
        return length == -1;
    }


    
}

File: dataStructure/queues/python/QueueWithLinkedlist.py
from Node import Node
class QueueWithLinkedlist:

    def __init__(self) -> None:
        self.length = -1
        self.first = None
        self.last = None

    def add(self, item):
        node = Node(item)

        if self.isEmpty():
            self.first = node
            self.last = node
            
        else:   
            node.next = self.last
            self.last = node
            
        self.length += 1
        
    def isEmpty(self):
        return self.length == -1
    

#tests
queue = QueueWithLinkedlist()

queue.add(10)
queue.add(20)
queue.add(30)
queue.add(40)
print("oi")